### DISCUSSION TITLE
Keycloak REST API Translation

### DISCUSSION THREAD

**Discussion by luan-baggio** [keycloak/keycloak#23815] [Q&A] [answered]
### Description

Hello. I'm currently setting up Keycloak as IAM and I cannot get translated API responses.

Is this not available? I've tried passing the locale header... But it didn't work
If it's not available, how is it done in the templates? Currently I'm mapping the error messages as keys in a local translation file. 

I know that you have the messages mapped by error types in themes, like "invalidUserMessage", etc. But how is the error mapping done?

**Comment by jonkoops** [comment, marked as answer]
Rest API responses are not localized. I'd recommend reading #16882 if you want to understand more about this.
